CMakeLists.txt

```diff
@@ -219,6 +219,9 @@ set(Launcher_SUBREDDIT_URL "https://prismlauncher.org/reddit" CACHE STRING "URL
 set(Launcher_FORCE_BUNDLED_LIBS OFF CACHE BOOL "Prevent using system libraries, if they are available as submodules")
 set(Launcher_QT_VERSION_MAJOR "6" CACHE STRING "Major Qt version to build against")
 
+# Java downloader
+option(ENABLE_JAVA_DOWNLOADER "Build the java downloader feature" ON)
```

Comment:
> By default, I set it to true, but I did not find a clean way to set it to false for Linux: can you suggest what I need to change for the cmake to turn it off for Linux in the cmake files?

this will cover BSD as well since distributions/packagers will have similar issues as discussed before

```suggestion
set(ENABLE_JAVA_DOWNLOADER_DEFAULT ON)

# Although we recommend enabling this, we cannot guarantee binary compatibility on 
# differing Linux/BSD/etc distributions. Downstream packagers should be explicitly opt-ing into this
# feature if they know it will work with their distribution.
if(UNIX AND NOT APPLE)
  set(ENABLE_JAVA_DOWNLOADER_DEFAULT OFF)
endif()

option(ENABLE_JAVA_DOWNLOADER "Build the java downloader feature" ON)
```

Reply:
this should be added to the configure step in CI though, as if our binaries work on a distribution, chances are one of the provided JREs will as well